### PR TITLE
pkg/api: Cleanup

### PR DESCRIPTION
**Pull request by orozery**
This PRremoves what's left of pkg/api.
This effect the gwctl/CRUD mode handling of policies.
The access policy API is retained, while the LB policy API is removed, as there are no more demos using it.

### CODE REVIEW

**Review by kfirtoledo — COMMENTED** [review]
LGTM

**Comment by elevran** [comment]
additional required checks would pass after rebase/update to latest main since the GH actions names on the branch protection rule have automatically changed to reflect the Go1.22 update.

**Comment by zivnevo** [comment]
@orozery , can you please resolve the conflicts?

File: cmd/gwctl/subcommand/policy.go
Comment (zivnevo) [inline cmd/gwctl/subcommand/policy.go:1]: Need to update command descriptions

**Review by zivnevo — CHANGES_REQUESTED** [review]
I'm not sure I understand how policy namespace is being handled.
Are we ignoring the namespace specified in the YAML manifest and always use the manager's namespace? Shouldn't we check that they are at least compatible? Is this consistent with the other CRDs?

File: pkg/controlplane/store/types.go [hunk 1]
```diff
@@ -88,30 +87,18 @@ func NewImport(imp *v1alpha1.Import) *Import {
...
-	Version uint32
-}
-
-// NewLBPolicy creates a new load-balancing policy.
-func NewLBPolicy(policy *api.Policy) *LBPolicy {
-	return &LBPolicy{
-		Policy:  *policy,
-		Version: accessPolicyStructVersion,
+		AccessPolicySpec: policy.Spec,
+		Name:             policy.Name,
```
[the remark this answers is not in the document]
Comment (orozery) [reply pkg/controlplane/store/types.go:101]: CRUD mode does not support namespaces.

**Review by zivnevo — APPROVED** [review]

**Pull request merged**